tools/sidekick/config.schema.json

```diff
@@ -0,0 +1,116 @@
+    "project": {
+      "type": "string",
+      "description": "The name of the project to display in the Helix Sidekick"
+    },
+    "host": {
+      "type": "string",
+      "format": "hostname",
+      "description": "The host name of the production website"
+    },
+    "container": {
+      "type": "string",
+      "description": "The ID of the dropdown container to add this plugin to"
+    },
+    "pushDown": {
+      "type": "boolean",
+      "description": "Have Helix Sidekick push down page content instead of overlaying it"
+    },
+    "pushDownSelector": {
+      "type": "string",
+      "description": "The query selector for absolute elements to also push down"
+    },
+    "plugins": {
+      "type": "array",
+      "items": { "$ref": "#/$defs/plugin" }
+    }
+   },
+   "additionalProperties": false,
+   "$defs": {
+    "plugin": {
+      "type": "object",
+      "properties": {
+        "container": {
+          "type": "string",
+          "description": "The ID of the container to add this plugin to"
+        },
+        "environments": {
+          "type": "array",
+          "items": {
+            "type": "string",
+            "enum": ["any", "edit", "preview", "live", "prod"]
+          },
+          "description": "The environments to display this plugin in (default: any)"
+        },
+        "event": {
+          "type": "string",
+          "description": "The name of a custom event to fire when the button is clicked"
+        },
+        "exclude_paths": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Exclude the plugin from these paths (glob patterns supported)"
+        },
+        "id": {
+          "type": "string",
+          "description": "The unique plugin ID"
+        },
+        "include_paths": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Include the plugin on these paths (glob patterns supported)"
+        },
+        "is_container": {
+          "type": "boolean",
+          "description": "Turns the plugin into a container for other plugins"
+        },
+        "title": {
+          "type": "string",
+          "description": "The button text"
+        },
+        "title_i18n": {
+          "type": "object",
+          "description": "The button text in other supported languages",
+          "properties": {
+            "de": {
+              "type": "string",
+              "description": "German"
+            },
+            "fr": {
+              "type": "string",
+              "description": "French"
+            }
+          },
+          "anyOf": [
+            { "required": [ "de" ] },
+            { "required": [ "fr" ] }
+          ],
+          "additionalProperties": false
+        },
+        "url": {
+          "type": "string",
+          "description": "The URL to open when the button is clicked"
+        }
+      },
+      "allOf": [
```

Comment:
Feels like the `allOf` could be dropped here and the result would be the same. Maybe I missing something about this constraint though?